src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -662,6 +662,50 @@ def info
       {'stemcell_formats' =>  ['vsphere-ovf', 'vsphere-ova']}
     end
 
+    #creates T1 router and virtual switch attached to it
+    def create_subnet(subnet_definition)
+      raise 'NSXT must be enabled in CPI to use create_subnet' if !@config.nsxt_enabled?
+      cloud_properties = subnet_definition['cloud_properties']
+      raise 'cloud_properties must be provided' if cloud_properties.nil?
+      raise 'edge_cluster_id cloud property can not be empty' if cloud_properties['edge_cluster_id'].nil?
+      raise 't0_router_id cloud property can not be empty' if cloud_properties['t0_router_id'].nil?
+      raise 'transport_zone_id cloud property can not be empty' if cloud_properties['transport_zone_id'].nil?
+      subnet = create_subnet_obj(subnet_definition['range'], subnet_definition['gateway'])
+
+      t1_router_id = nil
```

Comment:
These two variables aren't really necessary. It looks like they're here to facilitate the `rescue` clause. Either put each `@nsxt_provider` action into its own `begin ... rescue ... end` block or use something like:

```ruby
@nsxt_provider.delete_t1_router(t1_router.id) if defined? t1_router
```

Reply:
scope in ruby i believe is shared so you dont need to pull out there vars above, they will be defined from within begin and be available in rescue.

> if defined? t1_router

that wouldnt work, since defined? return "local-variable" always even if it's nil.

Reply:
You’re right. Looks like the check should be t1_router.nil?